### PR TITLE
[KubeRay] Documentation structure and skeleton

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

**You can take a look at the rendered docs outline by reading clicking _details_ on the _docs/readthedocs.org:ray_ item of the checks.**

This PR introduces the structure and outline for new KubeRay-based Ray-on-Kubernetes docs, in the Ray docs.
The PR in its current form does not introduce any new documentation content.

Currently, we have K8s docs two levels down from the root: 
`clusters/cluster managers/deploying on k8s.`

This PR places the new documentation at the top level of clusters:
`clusters/deploying on k8s`
with subtopics
`clusters/deploying on k8s/using gpus` etc.

The old docs stay where they are, with a note at the top explaining that KubeRay is preferred and links to KubeRay materials. The word "legacy" is introduced to the title of the old docs page.

## Ray docs vs. KubeRay docs

For the time being, we're choosing to include new docs in the Ray documentation, rather than in the KubeRay documentation.

The idea is to present users entering the Ray docs for the first time with a frictionless experience --
"Deploying on Kubernetes" is visible at the top level at docs.ray.io, and clicking the drop-down reveals all of the info necessary to get stuff done.

There's some risk of duplication of information between the two doc pages and decoherence of what information goes where. For now, we'll treat this as a secondary concern and focus on the documentation content.

The main goal for the Ray maintainers is for K8s users to be able to easily run their Ray workloads using the info in the Ray docs.  Target audience: non-specialists (not much experience with Ray, perhaps not much experience with K8s).

Some ideas for keeping the two doc sites coherent:

- There will be links to the KubeRay docs from the Ray docs. 
- Down the line, we might choose to embed "special topics"/"best practices" pages (using GPUs, logging) from the KubeRay docs into the Ray docs.

## Plan

Once we're aligned on doc structure, I can start implementing the individual pages, starting with the most basic info.
For ease of review, it's probably best to split this into multiple PRs.

### Sequencing doc PRs

~~Which of the following do people think is preferable?~~

Edit: We'll go with Option 1 below. 

**Please review the proposed documentation _structure_. On approval, I will merge this PR and proceed to implement individual pages in subsequent PRs.**

- **Option 1**
Merge this skeleton with some "under construction" pages in this PR, follow up with content in other PRs.
IMO, having some "under construction" pages is ok in the `master` docs, which most users do not access.
(docs.ray.io redirects to the latest docs, which reflect the latest Ray release.)

- ~~**Option 2**~~
~~Alternatively, I can remove the "under construction" pages and implement the index, quickstart, and configuration guide in this PR. Further PRs can then cover the rest of the content.~~

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kuberay.md
Comment (DmitriGekhtman) [inline doc/source/cluster/kuberay.md:1]: Re-purposing this as the index page and retitling from `Deploying with KubeRay` to `Ray on Kubernetes`.

File: doc/source/cluster/kuberay/kuberay-vs-legacy.md
Comment (DmitriGekhtman) [inline doc/source/cluster/kuberay/kuberay-vs-legacy.md:3]: There's naturally been a lot of user concern on this topic.
Definitely something we need to elucidate.

File: doc/source/cluster/kuberay/quickstart.md
Comment (DmitriGekhtman) [inline doc/source/cluster/kuberay/quickstart.md:3]: I've copy-pasted the content from the old `kuberay.md`.

Will revise in a follow-up.

File: doc/source/cluster/kuberay/quickstart.md
Comment (DmitriGekhtman) [inline doc/source/cluster/kuberay/quickstart.md:20]: Will get rid of this shell-script in a follow-up. The main purpose of this script was to check out a fixed KubeRay commit from the master branch.
The instructions will instead point to KubeRay 0.3.0 config resources.

File: doc/source/cluster/kubernetes.rst
Comment (DmitriGekhtman) [inline doc/source/cluster/kubernetes.rst:5]: Clearer wording here on the relative status of the two projects. This page is buried deeper than the new docs.

**Comment by richardliaw** [comment]
Let's go with option 1

**Review by pcmoritz — APPROVED** [review]

**Comment by pcmoritz** [comment]
@wuisawesome I think we need a CODEOWNER review from you here :)

**Comment by DmitriGekhtman** [comment]
> I am assuming here the white spaces are for the UI tree node names, not the actual documents or directories :) clusters/cluster managers/deploying on k8s

Yep, no whitespace in directory or file names.

**Comment by DmitriGekhtman** [comment]
Merging -- finer details can be resolved in future PRs.

**Pull request merged**
